services/app/src/pages/game/Game.jsx

```diff
@@ -8,19 +8,19 @@ import { useUserFromLocalStorage } from '../../hooks/useUserFromLocalStorage';
 import { WsContext } from '../../contexts/ws-context';
 
 function Room() {
-  const [roomUrl] = useState(window.location.href);
+  const [roomUrl] = useState(window.location.href.split('?')[0]);
   const { username } = useUserFromLocalStorage();
   const { users } = useWebSockets();
   const { ws } = useContext(WsContext);
 
   useEffect(() => {
     if (username) {
-      ws.send(username);
+      ws.onopen = () => ws.send(username);
     }
   }, [username]);
 
   const handleSubmit = () => {
-    ws.send(username);
+    ws.onopen = () => ws.send(username);
```

Comment:
Why are we adding this here?

Reply:
Solves "Failed to execute 'send' on 'WebSocket': Still in CONNECTING state" problem

Reply:
I would get that in the `useEffect`, but on `handleSubmit` you'd have to click the button before the connection is open.

Maybe a better way to handle would be to show something else than the game until the connection is open?

Reply:
Yes, but this was for a button logic, which is now is handled by useEffect, because after "login" modal, page is reloaded and connects to WS, from Custom Route, which checks local storage again for username